api/envoy/config/listener/v3/listener.proto

```diff
@@ -248,4 +249,9 @@ message Listener {
   // Configuration for :ref:`access logs <arch_overview_access_logs>`
   // emitted by this listener.
   repeated accesslog.v3.AccessLog access_log = 22;
+
+  // UdpWriterProto in Listener Config
+  // This field specifies the udp_writer_name, e.g. "udp_default_writer" or
+  // "udp_gso_batch_writer", that can be used to instantiate udp_writer.
```

Comment:
document that if this config is omitted, udp listener will use the default udp writer.

Reply:
Updated the comment here.

Reply:
From an end user perspective it's hard to understand how this would be used. Can you write the docs to be a bit more end user focused, with a description of what each built-in option does? Also, extensions load by proto message name, not friendly name, so it would be good to clarify that with examples if possible.

Also, please add a release note.

Reply:
> Also, please add a release note.

release note is here: docs/root/version_history/current.rst

Reply:
> extensions load by proto message name, not friendly name, so it would be good to clarify that with examples if possible.

Right now we are using friendly name to load:
```
 auto& config_factory =
        Config::Utility::getAndCheckFactoryByName<Network::UdpPacketWriterConfigFactory>(
            udp_writer_config.name());
```
The current two impls (batch writer and default writer) don't have their own config proto messages

Reply:
This is not correct. We now key on proto message name, so I would define an empty message for each if you need to and use the normal factory registration system.

Reply:
Updated the code to use `TypedFactory` and  using `getAndCheckFactory` to get the UdpPacketWriterConfigFactory.